BundleTransport/app/src/main/java/net/discdd/bundletransport/UsbFragment.java

```diff
@@ -0,0 +1,229 @@
+        getActivity().registerReceiver(mUsbReceiver, filter);
+
+        // Check initial USB connection
+        checkUsbConnection(3);
+
+        usbExchangeButton.setOnClickListener(v -> {
+            try {
+                populateUsb();
+            } catch (IOException e) {
+                logger.log(INFO, "Populate USB was unsuccessful");
+                throw new RuntimeException(e);
+            }
+            logger.log(INFO, "Sync button was hit");
+        });
+        return view;
+    }
+
+    /**
+     * Finds appropriate usb dir for ddd activities and copies device files.
+     *
+     * @throws IOException
+     */
+    private void populateUsb() throws IOException {
+        List<StorageVolume> storageVolumes = storageManager.getStorageVolumes();
+        for (StorageVolume volume : storageVolumes) {
+            if (volume.isRemovable() && !volume.isEmulated()) {
+                File usbStorageDir = volume.getDirectory();
+                if (usbStorageDir != null) {
+                    File dddTransportDir = new File(usbStorageDir, usbDirName);
+                    if (!dddTransportDir.exists()) {
+                        dddTransportDir = new File(usbStorageDir, "/DDD_transport/server");
+                        dddTransportDir.mkdirs();
+                        dddTransportDir = new File(usbStorageDir, "DDD_transport/client");
+                        dddTransportDir.mkdirs();
+                    }
+                    try {
+                        dddTransportDir = new File(usbStorageDir, "DDD_transport");
+                        copyFilesFromDevice(dddTransportDir);
+                    } catch (Exception e) {
+                        logger.log(WARNING, "copyFilesFromDevice failed to populateUsb");
+                        throw new RuntimeException("Bad call to copyFilesFromDevice", e);
+                    }
+                }
+                break;
+            }
+        }
+    }
+
+    /**
+     * Copies for-client files from transport device (handling downstream).
+     *
+     * @param dddTransportDir target directory; USBs directory
+     * @throws IOException
+     */
+    private void copyFilesFromDevice(File dddTransportDir) throws IOException, GeneralSecurityException, RoutingExceptions.ClientMetaDataFileException, InvalidKeyException {
+        List<Path> storageList;
+        Path devicePathForClient = transportPaths.toClientPath;
+        try (Stream<Path> walk = Files.walk(devicePathForClient)) {
+            storageList = walk.filter(Files::isRegularFile).collect(Collectors.toList());
+        }
+        if (storageList.isEmpty()) {
+            logger.log(INFO, "No bundles to download from device to USB");
+            logger.log(INFO, "Our empty storageList was " + devicePathForClient);
+            return;
+        }
+        for (Path deviceFilePath : storageList) {
+            Path targetPath = dddTransportDir.toPath().resolve(deviceFilePath.getFileName());
```

Comment:
you are missing the client subdirectory. perhaps you should call it the toClient subdirectory.